File: Visual Studio 2010/VBAzureBingMaps/WebRole1/WebRole.vb
Imports System.Collections.Generic
Imports System.Linq
Imports Microsoft.WindowsAzure
Imports Microsoft.WindowsAzure.Diagnostics
Imports Microsoft.WindowsAzure.ServiceRuntime

Public Class WebRole
    Inherits RoleEntryPoint
    Public Overrides Function OnStart() As Boolean

        ' 有关配置更改的信息处理
        ' 参见MSDN相关话题 http://go.microsoft.com/fwlink/?LinkId=166357.
        AddHandler RoleEnvironment.Changing, AddressOf RoleEnvironmentChanging

        Return MyBase.OnStart()
    End Function

    Private Sub RoleEnvironmentChanging(ByVal sender As Object, ByVal e As RoleEnvironmentChangingEventArgs)
        ' 如果配置设置改变
        If e.Changes.Any(Function(change) TypeOf change Is RoleEnvironmentConfigurationSettingChange) Then
            ' 设置e.Cancel为真重启角色实例
            e.Cancel = True
        End If
    End Sub
End Class

File: Visual Studio 2010/CSAzureBingMaps/WebRole1/WebRole.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Microsoft.WindowsAzure;
using Microsoft.WindowsAzure.Diagnostics;
using Microsoft.WindowsAzure.ServiceRuntime;

namespace AzureBingMaps.WebRole
{
    public class WebRole : RoleEntryPoint
    {
        public override bool OnStart()
        {
            DiagnosticMonitor.Start("DiagnosticsConnectionString");

            // 有关配置更改的信息处理
            // 参见MSDN相关话题 http://go.microsoft.com/fwlink/?LinkId=166357.
            RoleEnvironment.Changing += RoleEnvironmentChanging;

            return base.OnStart();
        }

        private void RoleEnvironmentChanging(object sender, RoleEnvironmentChangingEventArgs e)
        {
            // 如果配置设置改变
            if (e.Changes.Any(change => change is RoleEnvironmentConfigurationSettingChange))
            {
                // 设置e.Cancel为真重启角色实例
                e.Cancel = true;
            }
        }
    }
}
